From decc0d2efa2db7476faaa70043774b722317da78 Mon Sep 17 00:00:00 2001
From: Wim Taymans <wtaymans@redhat.com>
Date: Fri, 18 Sep 2026 14:58:01 +0200
Subject: [PATCH] audioconvert: handle latency passthrough on monitor ports

After we pass through the latency from the monitor ports to the
input ports, or from the input ports to the monitor ports, we are
done.

Only recalculate the port latency when we are not in passthrough.

Fixes #5393

diff --git a/spa/plugins/audioconvert/audioconvert.c b/spa/plugins/audioconvert/audioconvert.c
index 4d2947ea7..c0b5fa17d 100644
--- a/spa/plugins/audioconvert/audioconvert.c
+++ b/spa/plugins/audioconvert/audioconvert.c
@@ -3157,6 +3157,9 @@ static int port_set_latency(void *object,
 			this, direction, port_id, latency);
 
 	port = GET_PORT(this, direction, port_id);
+	if (port == NULL)
+		return -EINVAL;
+
 	if (latency == NULL) {
 		info = SPA_LATENCY_INFO(other);
 		have_latency = false;
@@ -3185,8 +3188,8 @@ static int port_set_latency(void *object,
 
 		if (oport != NULL)
 			port_update_latency(oport, &info, have_latency);
-	}
-	recalc_latencies(this, direction);
+	} else
+		recalc_latencies(this, direction);
 	return 0;
 }

